src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTORY = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private Node<K, V>[] table;
+    private int capacity;
+    private int size = 0;
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null) {
+            table = new Node[DEFAULT_INITIAL_CAPACITY];
+            capacity = DEFAULT_INITIAL_CAPACITY;
+        }
+        if (size > capacity * DEFAULT_LOAD_FACTORY) {
+            resize();
+        }
+        int keyHash = getHash(key);
+        if (table[keyHash % table.length] == null) {
+            table[keyHash % table.length] = new Node<>(key, value, null);
+        } else {
+            int checking = 0;
+            Node<K, V> newNode = table[keyHash % table.length];
+            while (newNode.next != null) {
+                if (Objects.equals(newNode.key, key)) {
+                    newNode.value = value;
+                    checking++;
+                    size--;
+                }
+                newNode = newNode.next;
+            }
+            if (checking != 1) {
+                if (Objects.equals(newNode.key, key)) {
+                    newNode.value = value;
+                    size--;
+                }
+                newNode.next = new Node<>(key, value, null);
+            }
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
+        for (Node<K, V> node : table) {
+            Node<K, V> current = node;
+            while (current != null) {
+                if (Objects.equals(current.key, key)) {
+                    return current.value;
+                }
+                current = current.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getHash(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode());
```

Comment:
If key == null -> return 0, but if key is not null - we should return key.hashCode() % current table length, as far as I remember.
```suggestion
        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
```